### PR TITLE
feat:remove-inline onclick handlers for validate button

### PR DESCRIPTION
**Pull request by NeetigyaPod**
### Testing done
mvn test shows no error. Screenshots attached.
Before:
image

After:
Screenshot from 2023-10-11 14-56-34

Screenshot from 2023-10-12 12-19-52

```[tasklist]
### Submitter checklist
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

Closes Issue: https://issues.jenkins.io/browse/JENKINS-72117
and https://issues.jenkins.io/browse/JENKINS-70529 (duplicate)
Remove onclick script according to the guidelines: https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers

### CODE REVIEW

File: src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly [hunk 1]
```diff
@@ -54,10 +54,11 @@
     <div style="float:right" class="ldap-validate-section">
       <span class="yui-button">
         <span>
-          <button id="${uid}" type="button" name="${name}" class="ldap-validate-button"
-                  onclick="ldapValidateButton('${descriptor.descriptorFullUrl}/${method}','${attrs.filter}',this,'${uid}'); return false;">
+        <span class="ldap-validate-button-reference-holder" data-id="${uid}" data-fullurl="${descriptor.descriptorFullUrl}/${method}" data-attributes="${attrs.filter}"/>
+          <button id="${uid}" type="button" name="${name}" class="ldap-validate-button">
             ${title}
           </button>
+          <st:adjunct includes = "jenkins.security.plugins.ldap.validation.validateButton"/>
```
Comment (Kevin-CB) [inline src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly:61]:
```suggestion
          <st:adjunct includes="jenkins.security.plugins.ldap.validation.validateButton"/>
```

File: src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly [hunk 2]
```diff
@@ -68,7 +69,9 @@
         <div id="${uid}_dialog">
           <d:invokeBody/>
           <f:entry>
-            <input type="submit" name="Validate" value="${attrs.submit}" class="ldap-validate submit-button primary" onclick="return false;"/>
+        <span class="ldap-validate-input-button-reference-holder"/>
+            <input id = "ldap-validate-input" type="submit" name="Validate" value="${attrs.submit}" class="ldap-validate submit-button primary"/>
```
Comment (Kevin-CB) [inline src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly:73]:
```suggestion
            <input id="ldap-validate-input" type="submit" name="Validate" value="${attrs.submit}" class="ldap-validate submit-button primary"/>
```

File: src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly [hunk 3]
```diff
@@ -68,7 +69,9 @@
...
+            <st:adjunct includes = "jenkins.security.plugins.ldap.validation.validateButtonOnClick"/>
```
Comment (Kevin-CB) [inline src/main/resources/jenkins/security/plugins/ldap/validation/validate.jelly:74]:
```suggestion
            <st:adjunct includes="jenkins.security.plugins.ldap.validation.validateButtonOnClick"/>
```

**Review by Kevin-CB — APPROVED** [review]
I tested it locally with the CSP plugin, it LGTM

**Comment by NeetigyaPod** [comment]
Removed the whitespaces

**Comment by NeetigyaPod** [comment]
@jtnord Can we please merge this before October end as I wish it to be part of my hacktoberfest contribution? Would appreciate a lot.

**Review by jtnord — APPROVED** [review]

**Comment by NeetigyaPod** [comment]
@jtnord Hi, can we please add the hacktoberfest topic in this repository? Or a label of hacktoberfest-accepted to this PR works as well.

**Pull request merged**